library/alloc/tests/lib.rs

```diff
@@ -40,6 +40,7 @@
 #![feature(panic_update_hook)]
 #![feature(slice_flatten)]
 #![feature(thin_box)]
+#![feature(str_internals)]
```

Comment:
Instead of making changes to `alloc`, it'd be better to just add a new file for these tests in `core/tests` like `library/core/tests/str_next_code_point.rs`. That way we can avoid adding more unstable features to alloc than necessary.